api/app/src/routes/medical/dtos/linkDTO.ts

```diff
@@ -1,4 +1,86 @@
+import { Person, getId } from "@metriport/commonwell-sdk";
+import { differenceBy } from "lodash";
+
 import { ExternalMedicalPartners } from "../../../external";
+import { Link } from "../schemas/link";
+import { GenderAtBirth } from "../../../models/medical/patient";
+import { PatientLinks } from "../schemas/link";
 
 export type PatientLinkStatusDTO = "linked" | "needs-review";
 export type PatientLinksDTO = { [k in ExternalMedicalPartners]: PatientLinkStatusDTO };
+
+export function dtoFromCW({
+  cwPotentialPersons,
+  cwCurrentPersons,
+}: {
+  cwPotentialPersons: Person[];
+  cwCurrentPersons: Person[];
+}): PatientLinks {
+  let potentialLinks: Link[] = [];
+  const currentLinks: Link[] = [];
+
+  cwPotentialPersons.forEach(person => {
+    const personLink = convertFromCWPersonToLink(person);
+
+    if (personLink) potentialLinks.push(personLink);
+  });
+
+  cwCurrentPersons.forEach(person => {
+    const personLink = convertFromCWPersonToLink(person);
+
+    if (personLink) currentLinks.push(personLink);
+  });
+
+  if (currentLinks.length) {
+    const removePotentialLinksDuplicates = differenceBy(potentialLinks, currentLinks, "entityId");
+
+    potentialLinks = removePotentialLinksDuplicates;
+  }
+
+  return {
+    potentialLinks,
+    currentLinks,
+  };
+}
+
+export function convertFromCWPersonToLink(person: Person): Link | null {
+  const personId = getId(person);
+
+  const address = person.details?.address?.length ? person.details?.address[0] : undefined;
+  const personName = person.details?.name?.length ? person.details?.name[0] : undefined;
+
+  if (personId) {
+    const personLink: Link = {
+      entityId: personId,
+      potential: true,
+      source: ExternalMedicalPartners.COMMONWELL,
+      patient: {
+        id: personId,
+        firstName: personName && personName.given?.length ? personName.given[0] : "",
+        lastName: personName && personName.family?.length ? personName.family[0] : "",
+        dob: person.details?.birthDate ? person.details.birthDate : "",
+        genderAtBirth: displayGender(person),
+        personalIdentifiers: [],
+        address: {
+          addressLine1: address && address.line ? address.line[0] : "",
+          city: address && address.city ? address.city : "",
+          state: address && address.state ? address.state : "",
+          zip: address && address.zip ? address.zip : "",
+          country: address && address.country ? address.country : "",
+        },
```

Comment:
Could reuse `makePersonForPatient()`, on `patient-conversion.ts`. If we need a tweak and that function doesn't fit this use case, that file is a great place to store patient-related conversion functions.

Talked about this previously [here](https://github.com/metriport/metriport/pull/135#discussion_r1128417918).

Reply:
This seems different though because we are converting a person to our definition of a link